src/Storefront/Resources/app/storefront/src/plugin/header/cart-widget.plugin.js

```diff
@@ -40,27 +41,31 @@ export default class CartWidgetPlugin extends Plugin {
      * and persist the response to the browser's session storage
      */
     fetch() {
-        this._client.get(window.router['frontend.checkout.info'], (content, response) => {
-            if (response.status >= 500) {
-                return;
-            }
-
-            if (response.status === 204) {
-                Storage.removeItem(this.options.cartWidgetStorageKey);
-                const emptyCartWidget = Storage.getItem(this.options.emptyCartWidgetStorageKey);
-                if (emptyCartWidget) {
-                    this.el.innerHTML = emptyCartWidget;
+        fetch(window.router['frontend.checkout.info'], { method: 'GET', headers: { 'X-Requested-With': 'XMLHttpRequest' } })
```

Comment:
using that in the client is no option? because method and headers are always the same in your changes :thinking:

Reply:
thanks for explaining :slightly_smiling_face: